qml/pages/LoginPage.qml

```diff
@@ -21,42 +21,117 @@ Page {
 
         onAccessTokenFail: {
             console.log('access token failed')
-            webView.visible = true
         }
     }
 
-    SilicaWebView {
-        id: webView
-        anchors.fill: parent
-        url: page.startUrl + "&state=" + page.processId
+    Slack.AuthServer {
+        id: server
 
-        header: PageHeader {
-            title: "Sign in with Slack"
+        onResultUrlAvailable: {
+            authenticator.fetchAccessToken(url);
         }
+    }
 
-        onNavigationRequested: {
-            if (isReturnUrl(request.url)) {
-                visible = false
-                request.action = WebView.IgnoreRequest
+    Connections {
+        target: server
+        onListening: {
+            listening.text = (server.isListening() ? qsTr("Listening on port %1") : qsTr("Not listening on port %1")).arg(3000)
+        }
+    }
 
-                if (isSuccessUrl(request.url)) {
-                    authenticator.fetchAccessToken(request.url)
+    onStatusChanged: {
+        switch (status) {
+        case PageStatus.Active:
+            server.listen(3000);
+            Qt.openUrlExternally(page.startUrl + "&state=" + page.processId);
+            break;
+        case PageStatus.Deactivating:
+            server.close();
+            break;
+        }
+    }
+
+    SilicaFlickable {
+        anchors.fill: parent
+
+        contentWidth: column.width
+        contentHeight: column.height
+
+        Column {
+            id: column
+            width: page.width
+            spacing: Theme.paddingLarge
+
+            PageHeader {
+                title: "Sign in with Slack"
+            }
+
+            Label {
+                id: listening;
+                font.italic: true
+                wrapMode: Text.Wrap
+            }
+
+            Label {
+                text: qsTr("You can login with your browser:")
+            }
+
+            Button {
+                text: qsTr("Use browser")
+                onClicked: {
+                    Qt.openUrlExternally(page.startUrl + "&state=" + page.processId);
+                }
+            }
+
+            Label {
+                text: qsTr("Or use the Sailfish WebView:")
+            }
+
+            Button {
+                text: qsTr("Use webview")
+                onClicked: {
+                    // Closing the 3000 port as even though the webview would successfully connect to it, the timing it navigates away from this page makes
+                    // the webview crash (at onResultUrlAvailable signal).
+                    server.close();
```

Comment:
Without the server running, the login via webview never finishes for me.

To avoid the crash, you should close the client socket after sending the response HTML in AuthServer, then the webview no longer crashes, because the connection is terminated before the webview is destroyed by the pagestack pop.

Reply:
I'll investigate this more. The first commit of this branch there wass no problem with the webivew connecting locally (it was actually the only way to get back the token)

Reply:
I think I got this: the webview was a bit more picky about my "http" response, and it was still 'loading' when destroyed. Adding Content-Length to the response seems to have fixed that. Let me know if this works on your end.